src/data/mv_polynomial.lean

```diff
@@ -1407,6 +1418,54 @@ lemma pderivative_zero {v : S} : pderivative v (0 : mv_polynomial S R) = 0 :=
 suffices pderivative v (C 0 : mv_polynomial S R) = 0, by simpa,
 show pderivative v (C 0 : mv_polynomial S R) = 0, from pderivative_C
 
+section
+variables (R)
+
+/-- `pderivative : S → mv_polynomial S R → mv_polynomial S R` as an `add_monoid_hom`  -/
+def pderivative.add_monoid_hom (v : S) : mv_polynomial S R →+ mv_polynomial S R :=
+{ to_fun := pderivative v,
+  map_zero' := pderivative_zero,
+  map_add' := λ x y, pderivative_add, }
```

Comment:
@kbuzzard, did someone ever write your `collectible` tactic? It seems perfect here.

Reply:
What got written is [this](https://github.com/ImperialCollegeLondon/natural_number_game/blob/master/src/tactic/structure_helper.lean). [here](https://github.com/ImperialCollegeLondon/natural_number_game/blob/4c6ff5088b33bfe49d3fc068ae713045317e4338/src/game/world2/level6.lean#L83) is an example of usage. If I have already defined `mynat.zero` and `mynat.add_comm` etc -- every single field -- then `instance : add_comm_monoid mynat := by structure_helper` makes the instance.

Reply:
Maybe it's too ambitious to hope this can be assembled automatically at this point. :-)